bytecomp/simplif.ml

```diff
@@ -460,14 +460,14 @@ let simplify_lets lam =
   | Llet(str, v, Lvar w, l2) when optimize ->
       Hashtbl.add subst v (simplif (Lvar w));
       simplif l2
-  | Llet(Strict, v, Lprim(Pmakeblock(0, Mutable), [linit]), lbody)
-    when optimize ->
+  | Llet(Strict, v, Lprim(Pmakeblock(0, Mutable, shape), [linit]), lbody)
+    when optimize && not !Clflags.native_code ->
```

Comment:
Why do you disable turning local refs into mutable variable for native code?  Is it because it is handled elsewhere?  Would it harm to leave it there?